burn-wgpu/src/context/base.rs

```diff
@@ -97,6 +107,16 @@ impl Context {
         pipeline: Arc<ComputePipeline>,
         buffers: &[&Buffer],
     ) {
+        if !self.is_tuning.load(Ordering::Relaxed) && !self.tuning_template_ids.lock().is_empty() {
+            // clean cache of pipelines accumulated during tuning
+            let mut cache = self.cache.lock();
+            let mut tuning_template_ids = self.tuning_template_ids.lock();
+            for template_id in tuning_template_ids.iter() {
+                cache.remove(template_id);
+            }
+            tuning_template_ids.clear();
+        }
+
```

Comment:
Looking at the code, I think we can avoid locking when we aren't tuning; we can move this code in the `stop_tuning` function, just cleaning the cache right after tuning!

Reply:
Good point, I'll make the fix.